### PR TITLE
BP-2283-paypereemail-including-creditmanagement-are-not-showing-the-middle-name

### PR DESCRIPTION
**Pull request by abeka-buckaroo**

### CODE REVIEW

File: Model/Method/PayPerEmail.php [hunk 1]
```diff
@@ -206,6 +208,9 @@ private function getPayperemailService($payment)
         /** @var \Buckaroo\Magento2\Model\ConfigProvider\Method\PayPerEmail $config */
         $config = $this->configProviderMethodFactory->get('payperemail');
         $storeId = $payment->getOrder()->getStoreId();
+        /** @var \Buckaroo\Magento2\Model\ConfigProvider\Method\PayPerEmail $config */
+
+        $middleName = $this->_checkoutSession->getQuote()->getBillingAddress()->getMiddlename();
```
Comment (LucianTuriacArnia) [inline Model/Method/PayPerEmail.php:213]: To get the middleName is better to get it from order object from the payment, $payment->getOrder()->getBillingAddress()->getMiddlename(). Also CheckoutSession is resource-intensive class that should be instatiated on constructor, you should do it only as a Proxy.

**Review by LucianTuriacArnia — DISMISSED** [review]

**Comment by harli91** [comment]
I think this is not the correct way to do it, we get the first name and the last name from custom fields both on the frontend and admin area, we can end up with a combination of first name/last name from the custom fields and a middle name from the billing address

**Comment by Buckaroo-Rens** [comment]
@abeka-buckaroo Can you also add the ticket number in front of the PR name. This is standard procedure.

File: Model/Method/PayPerEmail.php [hunk 2]
```diff
@@ -206,7 +205,8 @@ private function getPayperemailService($payment)
         /** @var \Buckaroo\Magento2\Model\ConfigProvider\Method\PayPerEmail $config */
         $config = $this->configProviderMethodFactory->get('payperemail');
         $storeId = $payment->getOrder()->getStoreId();
-
+        /** @var \Buckaroo\Magento2\Model\ConfigProvider\Method\PayPerEmail $config */
```
Comment (LucianTuriacArnia) [inline Model/Method/PayPerEmail.php:208]: remove this

**Review by LucianTuriacArnia — DISMISSED** [review]

**Review by LucianTuriacArnia — DISMISSED** [review]

**Review by harli91 — APPROVED** [review]

**Pull request merged**